lsif-semanticdb/src/main/java/com/sourcegraph/lsif_semanticdb/LsifSemanticdb.java

```diff
@@ -127,21 +124,27 @@ private Integer processDocument(
         }
 
         // Hover
-        ArrayList<MarkedString> markedStrings = new ArrayList<>();
+        ArrayList<String> markupContents = new ArrayList<>(2);
         String documentation = symbolInformation.getDocumentation().getMessage();
         if (!documentation.isEmpty()) {
-          markedStrings.add(new MarkedString(Semanticdb.Language.UNKNOWN_LANGUAGE, documentation));
+          documentation = documentation.replaceAll("\n", "\n\n");
+          markupContents.add(documentation);
         }
 
         if (symbolInformation.hasSignature()) {
-          markedStrings.add(
-              new MarkedString(
-                  doc.semanticdb.getLanguage(),
-                  new SignatureFormatter(symbolInformation, symtab).formatSymbol()));
+          String b =
+              "```"
+                  + doc.semanticdb.getLanguage().toString().toLowerCase(Locale.ROOT)
+                  + '\n'
+                  + new SignatureFormatter(symbolInformation, symtab).formatSymbol()
```

Comment:
What do yo uthink about rendering the markup in `SignatureFormatter`? This is a hot path and we should try to to allocate as few intermediary strings as possible. It's best to use only one stringbuilder

Reply:
Ideally, we want to generate the markup content directly from a single string builder.

Reply:
Id be interested in profiling for sure, Im not confident we'll see a notably improvement, given that calling toString on a StringBuilder causes a copy _anyways_

Reply:
You're probably right. Ignore my comment. I personally try to stick with one stringbuilder per emitted string in hot paths, but I've never validated the performance implications (short-lived allocations on the JVM are surprisingly cheap)